src/worker/tasks/wallet-subscription-worker.ts

```diff
@@ -0,0 +1,168 @@
+import { type Job, type Processor, Worker } from "bullmq";
+import { getAllWalletSubscriptions } from "../../shared/db/wallet-subscriptions/get-all-wallet-subscriptions";
+import { getConfig } from "../../shared/utils/cache/get-config";
+import { logger } from "../../shared/utils/logger";
+import { redis } from "../../shared/utils/redis/redis";
+import { WalletSubscriptionQueue } from "../queues/wallet-subscription-queue";
+import { logWorkerExceptions } from "../queues/queues";
+import { SendWebhookQueue } from "../queues/send-webhook-queue";
+import { WebhooksEventTypes } from "../../shared/schemas/webhooks";
+import { getChain } from "../../shared/utils/chain";
+import { thirdwebClient } from "../../shared/utils/sdk";
+import { getWalletBalance } from "thirdweb/wallets";
+import type { Chain } from "thirdweb/chains";
+import type { WalletCondition } from "../../shared/schemas/wallet-subscription-conditions";
+import type { Webhooks } from "@prisma/client";
+
+interface WalletSubscriptionWithWebhook {
+  id: string;
+  chainId: string;
+  walletAddress: string;
+  conditions: WalletCondition[];
+  webhookId: number | null;
+  webhook: Webhooks | null;
+  createdAt: Date;
+  updatedAt: Date;
+  deletedAt: Date | null;
+}
+
+// Split array into chunks of specified size
+function chunk<T>(arr: T[], size: number): T[][] {
+  return Array.from({ length: Math.ceil(arr.length / size) }, (_, i) =>
+    arr.slice(i * size, i * size + size),
+  );
+}
+
+/**
+ * Verify if a condition is met for a given wallet
+ * Returns the current value if condition is met, undefined otherwise
+ */
+async function verifyCondition({
+  condition,
+  walletAddress,
+  chain,
+}: {
+  condition: WalletCondition;
+  walletAddress: string;
+  chain: Chain;
+}): Promise<string | undefined> {
+  switch (condition.type) {
+    case "token_balance_lt":
+    case "token_balance_gt": {
+      const currentBalanceResponse = await getWalletBalance({
+        address: walletAddress,
+        client: thirdwebClient,
+        tokenAddress:
+          condition.tokenAddress === "native"
+            ? undefined
+            : condition.tokenAddress,
+        chain,
+      });
+
+      const currentBalance = currentBalanceResponse.value;
+      const threshold = BigInt(condition.value);
+
+      const isConditionMet =
+        condition.type === "token_balance_lt"
+          ? currentBalance < threshold
+          : currentBalance > threshold;
+
+      return isConditionMet ? currentBalance.toString() : undefined;
+    }
+    default: {
+      // For TypeScript exhaustiveness check
+      const _exhaustiveCheck: never = condition;
+      return undefined;
+    }
```

Comment:
You shouldn't need this if you type `condition.type` properly. Typescript knows theres no unhandled cases.